Quorum/apis/llms/llm_conversation.py

```diff
@@ -0,0 +1,105 @@
+from enum import StrEnum, auto
+from pathlib import Path
+
+class MessageRole(StrEnum):
+    USER = auto()
+    ASSISTANT = auto()
+
+
+MAX_TOKENS = 1024
+
+
+def tmp_send_message_to_claude(client, model, messages):
+    try:
+        response = client.messages.create(
+            max_tokens=MAX_TOKENS,
+            model=model,
+            messages=messages
+        )
+    except:
+        raise
+
+    with open(Path(__file__).parent / 'ClaudeExecutionsNum.txt', 'r+') as f:
+        num = int(f.read())
+        f.seek(0)
+        f.write(str(num + 1))
+        f.truncate()
+
+    return response
+
+
+class LLMConversation:
+    '''
+    Description
+    -----------
+    At least GPT and Claude share the exact same API. Probably other LLMs use the same API as well,
+    just using different clients. This abstract class builds is expected to be inherited and build
+    the relevant client to be used with the same implementation of interacting with the LLM while 
+    maintaining a conversation context.
+    '''
+    def __init__(self):
+        '''
+        Summary
+        -------
+        Initialize a conversation manager for an LLM.
+        '''
+        self.client = None
+        self.model = None
+        self.messages: list[dict[str, str]] = []
```

Comment:
Class variable

Reply:
This is instance specific.

Reply:
Do we have more than one instance?

Reply:
Currently no. We might have in the future and it helped me when testing actually.